operators/pkg/controller/elasticsearch/nodespec/resources.go

```diff
@@ -0,0 +1,65 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package nodespec
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	commonv1alpha1 "github.com/elastic/cloud-on-k8s/operators/pkg/apis/common/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/settings"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/sset"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/version"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+)
+
+// Resources contain per-NodeSpec resources to be created.
+type Resources struct {
+	StatefulSet     appsv1.StatefulSet
+	HeadlessService corev1.Service
+	Config          settings.CanonicalConfig
+	// TLS certs
```

Comment:
Was a placeholder for future per-statefulset TLS certs resources to reconcile. But I'm going to remove that since we're not sure TLS certs will be reconciled this way.